.github/workflows/_build_bodo_pip.yml

```diff
@@ -87,10 +87,10 @@ jobs:
             echo "NEW_PYTHON_DIR: $NEW_PYTHON_DIR"
             rm -rf $OLD_PYTHON_DIR
             ln -s $NEW_PYTHON_DIR $OLD_PYTHON_DIR
-          CIBW_ENVIRONMENT: DISABLE_CCACHE=1 SCCACHE_BUCKET=engine-codebuild-cache SCCACHE_REGION=us-east-2 SCCACHE_S3_USE_SSL=true SCCACHE_S3_SERVER_SIDE_ENCRYPTION=true MACOSX_DEPLOYMENT_TARGET=${{ inputs.os == 'macos-14' && '11.0' || '10.15' }} BODO_VENDOR_MPICH=1
+          CIBW_ENVIRONMENT: DISABLE_CCACHE=1 SCCACHE_BUCKET=engine-codebuild-cache SCCACHE_REGION=us-east-2 SCCACHE_S3_USE_SSL=true SCCACHE_S3_SERVER_SIDE_ENCRYPTION=true MACOSX_DEPLOYMENT_TARGET=${{ inputs.os == 'macos-14' && '12.0' || '10.15' }} BODO_VENDOR_MPICH=1
```

Comment:
As discussed in https://bodoai.slack.com/archives/CP7KK34SD/p1732731734692499, we need to bump the min version to MacOS 12.